refactor(formatter): check whether BlockIndent's content is empty in debug mode

Dunqing opened the pull request:
Using `snapshot` to handle empty content is not ideal because it has a slight overhead. Additionally, it should report immediately that there is empty content rather than silencing it, so that we can improve the code to avoid unnecessary logic.

leaysgur reviewed (approved):
Please let me know if my understanding is wrong:

We can remove snapshot here because empty check is already done before.

Dunqing committed:
refactor(formatter): check whether BlockIndent's content is empty in debug mode (#16437)

Using `snapshot` to handle empty content is not ideal because it has a slight overhead. Additionally, it should report immediately that there is empty content rather than silencing it, so that we can improve the code to avoid unnecessary logic.
